### PR TITLE
Implements the long-timeout module to get past the 2147483647 ms limitation

### PR DESCRIPTION
**Pull request by reedog117**
Addresses #61 by implementing the ```long-timeout``` module.

### CODE REVIEW

**Comment by arb** [comment]
Please use https://www.npmjs.com/package/big-time instead as it has unit tests and is version 1.

**Comment by Marsup** [comment]
Can't we do better than Math.min? Mocking moment?

File: package.json [hunk 1]
```diff
@@ -9,6 +9,7 @@
     "test-cov-html": "lab -m 5000 -r html -o coverage.html -La code"
   },
   "dependencies": {
+    "big-time": "^1.0.2",
```
Comment (arb) [inline package.json]: 1.0.x

**Comment by arb** [comment]
Also, can you please add some kind of test showing what the expected behavior is when there is a long timeout?

File: lib/index.js [hunk 2]
```diff
@@ -27,29 +28,31 @@ var internals = {
         hourly: 'hour',
         daily: 'day',
         weekly: 'week',
-        monthly: 'monthly',
+        monthly: 'month',
         yearly: 'year'
     },
     sanitize: new RegExp(Hoek.escapeRegex(Path.sep), 'g'),
     MAX_TIME: 2147483647 // 2^31-1
```
Comment (arb) [inline lib/index.js]: I don't think we need this anymore.

File: lib/index.js [hunk 3]
```diff
@@ -27,29 +28,31 @@ var internals = {
...
+    var timeout;
 
-    // match [1] user string
+    period = period.toLowerCase();
     now.endOf(internals.timeMap[period]);
+    timeout = now.valueOf() - Date.now();
 
-    var timeout = now.valueOf() - Date.now();
-    timeout = Math.min(timeout, internals.MAX_TIME);
+    timeout = Math.min(timeout, timeout); // for test compatibility
```
Comment (arb) [inline lib/index.js]: Shouldn't need this anymore.

File: lib/index.js (hunk 2 above)
Comment (reedog117) [inline lib/index.js]: This was a typo I introduced that I fixed after the fact.

**Comment by reedog117** [comment]
Weird -- my code is passing tests with the same iojs version locally.

File: lib/index.js (hunk 2 above)
Comment (arb) [inline lib/index.js]: I mean remove `MAX_TIME` since it's not being used anymore.

File: lib/index.js [hunk 4]
```diff
@@ -134,8 +135,10 @@ module.exports = internals.GoodFile = function (events, config) {
         squeeze: Squeeze(events),
         stringify: SafeJson(null, { separator: '\n' })
     };
-};
 
+    this._momentInstance = Moment;
```
Comment (arb) [inline lib/index.js]: No. Require these libraries in the tests and override the methods there.

File: test/index.js [hunk 5]
```diff
@@ -285,7 +285,7 @@ describe('GoodFile', function () {
         });
     });
 
-    it('rotates logs on the specified internal', function (done) {
+    it('rotates logs on the specified interval', function (done) {
```
Comment (arb) [inline test/index.js]: This test shouldn't require this many changes... You should only need to update some of the tests. You need to add a new test for large timeouts.

File: lib/index.js (hunk 4 above)
Comment (reedog117) [inline lib/index.js]: How do I do that if the library instances within GoodFile aren't public, and those variables are closured? I don't think it's like the old test which overrode _Math_ which is a global library so changing it in the test changes it in GoodFile.

File: test/index.js [hunk 6]
```diff
@@ -57,10 +60,29 @@ internals.readStream = function (done) {
 
 var describe = lab.describe;
 var it = lab.it;
+var before = lab.before;
 var expect = Code.expect;
 
 describe('GoodFile', function () {
 
+    before(function (done) {
```
Comment (arb) [inline test/index.js]: We avoid `before` in test code.

File: lib/index.js (hunk 2 above)
Comment (arb) [inline lib/index.js]: `MAX_TIME` is still present...?

File: test/index.js [hunk 7]
```diff
@@ -302,12 +305,117 @@ describe('GoodFile', function () {
...
+                for (var i = 0; i < 10; ++i) {
+                    read.push({ event: 'request', statusCode: 200, id: i, tag: 'my test 2 - ' + i });
+                }
+
+                read.push(null);
+            }, 175);
+        });
+    });
+
+    it('correctly handles large timeouts', function (done) {
```
Comment (arb) [inline test/index.js]: This doesn't really test the long timeout code... though now that I'm thinking about it, we know that big-time works so we can remove this test.

**Comment by arb** [comment]
@reedog117 one final comment, then can you please rebase this PR so it's one or two commits instead of 11? Thanks!

File: test/index.js (hunk 7 above)
Comment (reedog117) [inline test/index.js]: I thought that changing _TIMEOUT_MAX to something very small would force big-time to create successive timeouts rather than 1 bigger timeout. I also temporarily added a few console.log() statements to doublecheck that that is actually the case.

Comment (arb) [inline test/index.js]: Yes that is what is supposed to happen. I guess what I'm saying is that because we know how big-time works, because of *it's* tests, we don't really need to test this edge case. Earlier good-file tests show that the files get created and named properly at the designed interval. All this is doing is exercising big-time code that we know has already been tested and confirmed to work.

You can leave it in though if you think it adds value to the module.

**Comment by arb** [comment]
Need to pull from upstream; can't automerge.

**Comment by reedog117** [comment]
I'll fix it once I'm back from vacation.
